src/adapter/etl-adapter-excel/src/Flow/ETL/Adapter/Excel/ExcelExtractor.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flow\ETL\Adapter\Excel;
+
+use function Flow\ETL\DSL\array_to_rows;
+use Flow\ETL\Extractor\{Limitable, LimitableExtractor, Signal};
+use Flow\ETL\{Extractor, FlowContext, Loader\Closure};
+use Flow\Filesystem\Exception\InvalidArgumentException;
+use Flow\Filesystem\Path;
+use OpenSpout\Common\Entity\{Cell, Row};
+use OpenSpout\Reader\ODS\Reader as OdsReader;
+use OpenSpout\Reader\XLSX\Reader as XlsxReader;
+
+final class ExcelExtractor implements Closure, Extractor, LimitableExtractor
+{
+    use Limitable;
+
+    private XlsxReader|OdsReader $reader;
+
+    private bool $withHeader = true;
+
+    public function __construct(private readonly Path $path)
+    {
+        $this->resetLimit();
+
+        $this->reader = match ($this->path->extension()) {
+            'xlsx' => new XlsxReader(),
+            'ods' => new OdsReader(),
+            default => throw new InvalidArgumentException('Unsupported file extension: ' . ($this->path->extension() ?: 'n/a')),
```

Comment:
Sometimes files might not have extension but they are still valid excel files and that wont work for them, what I would do here is introduce enum `ExcelReader` with two options `XLSX` and `ODS` and I would allow to set through `withReader(ExcelReader $type) : self` 

Then I would make a property: 

```
private null|XlsxReader|OdsReader $reader;
```

and a private method `private function reader() : XlsxReader|OdsReader` that would either: 

- create the one that user passed
- detect it from path extension
- throw an error when user set nothing and there is no extension